### ISSUE TITLE
Unable to pip install comet-ml in a pytorch docker image

### ISSUE THREAD

**Issue by david-wb** [comet-ml/issue-tracking#395]
```
FROM pytorch/pytorch:1.6.0-cuda10.1-cudnn7-runtime

RUN pip install comet-ml
```

Doesn't work. Will be forced to switch to wandb until this is resolved. Thanks!

**Comment by dsblank** [comment]
Maybe because the package is named `comet_ml` ?

**Comment by dsblank** [comment]
PS - I did my PhD at IU, finishing a few years before you got there (1997). I bet it is cold in Bloomington right now :)

**Comment by dsblank** [comment]
Modern pips should be able to install by `comet-ml` too. Can you provide more details:

* full error
* environment (python version, pip version)
* os

**Comment by david-wb** [comment]
Hi @dsblank I live in PC, Utah now

> PS - I did my PhD at IU, finishing a few years before you got there (1997). I bet it is cold in Bloomington right now :)

Hah, I bet it is but I live in Park City, Utah and it's even colder!

**Comment by david-wb** [comment]
Here's the full error log. To reproduce simply try building the above Dockerfile.

```
$ docker build -t eltest:latest .
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM pytorch/pytorch:1.6.0-cuda10.1-cudnn7-runtime
1.6.0-cuda10.1-cudnn7-runtime: Pulling from pytorch/pytorch
23884877105a: Already exists 
bc38caa0f5b9: Already exists 
2910811b6c42: Already exists 
36505266dcc6: Already exists 
3472d01858ba: Pull complete 
4a98b57681ff: Pull complete 
f3b419d1e6d5: Pull complete 
Digest: sha256:9c3aa4653f6fb6590acf7f49115735be3c3272f4fa79e5da7c96a2c901631352
Status: Downloaded newer image for pytorch/pytorch:1.6.0-cuda10.1-cudnn7-runtime
 ---> 6a2d656bcf94
Step 2/2 : RUN pip install comet-ml
 ---> Running in 946c59773728
Collecting comet-ml
  Downloading comet_ml-3.3.3-py2.py3-none-any.whl (250 kB)
Requirement already satisfied: six in /opt/conda/lib/python3.7/site-packages (from comet-ml) (1.14.0)
Collecting wurlitzer>=1.0.2
  Downloading wurlitzer-2.0.1-py2.py3-none-any.whl (6.1 kB)
Collecting requests-toolbelt>=0.8.0
  Downloading requests_toolbelt-0.9.1-py2.py3-none-any.whl (54 kB)
Collecting jsonschema!=3.1.0,>=2.6.0
  Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting wrapt>=1.11.2
  Downloading wrapt-1.12.1.tar.gz (27 kB)
Requirement already satisfied: requests>=2.18.4 in /opt/conda/lib/python3.7/site-packages (from comet-ml) (2.23.0)
Collecting nvidia-ml-py3>=7.352.0
  Downloading nvidia-ml-py3-7.352.0.tar.gz (19 kB)
Collecting everett[ini]>=1.0.1; python_version >= "3.0"
  Downloading everett-1.0.3-py2.py3-none-any.whl (31 kB)
Collecting dulwich>=0.20.6; python_version >= "3.0"
  Downloading dulwich-0.20.19-cp37-cp37m-manylinux2010_x86_64.whl (508 kB)
Collecting netifaces>=0.10.7
  Downloading netifaces-0.10.9.tar.gz (28 kB)
Collecting websocket-client>=0.55.0
  Downloading websocket_client-0.57.0-py2.py3-none-any.whl (200 kB)
Requirement already satisfied: setuptools in /opt/conda/lib/python3.7/site-packages (from jsonschema!=3.1.0,>=2.6.0->comet-ml) (46.4.0.post20200518)
Collecting attrs>=17.4.0
  Downloading attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting pyrsistent>=0.14.0
  Downloading pyrsistent-0.17.3.tar.gz (106 kB)
Collecting importlib-metadata; python_version < "3.8"
  Downloading importlib_metadata-3.4.0-py3-none-any.whl (10 kB)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /opt/conda/lib/python3.7/site-packages (from requests>=2.18.4->comet-ml) (1.25.8)
Requirement already satisfied: chardet<4,>=3.0.2 in /opt/conda/lib/python3.7/site-packages (from requests>=2.18.4->comet-ml) (3.0.4)
Requirement already satisfied: idna<3,>=2.5 in /opt/conda/lib/python3.7/site-packages (from requests>=2.18.4->comet-ml) (2.9)
Requirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.7/site-packages (from requests>=2.18.4->comet-ml) (2020.6.20)
Collecting configobj; extra == "ini"
  Downloading configobj-5.0.6.tar.gz (33 kB)
Collecting typing-extensions>=3.6.4; python_version < "3.8"
  Downloading typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
Collecting zipp>=0.5
  Downloading zipp-3.4.0-py3-none-any.whl (5.2 kB)
Building wheels for collected packages: wrapt, nvidia-ml-py3, netifaces, pyrsistent, configobj
  Building wheel for wrapt (setup.py): started
  Building wheel for wrapt (setup.py): finished with status 'done'
  Created wheel for wrapt: filename=wrapt-1.12.1-py3-none-any.whl size=19553 sha256=1504f59f68aae55b3df1f4ceb59be366d9b1e3d1a3eb0cd5b0f57c7eac572c3a
  Stored in directory: /root/.cache/pip/wheels/62/76/4c/aa25851149f3f6d9785f6c869387ad82b3fd37582fa8147ac6
  Building wheel for nvidia-ml-py3 (setup.py): started
  Building wheel for nvidia-ml-py3 (setup.py): finished with status 'done'
  Created wheel for nvidia-ml-py3: filename=nvidia_ml_py3-7.352.0-py3-none-any.whl size=19189 sha256=2afec49b6c826b0c19fac451047b973d4f0ce2235d265a4213fcabfde199a95b
  Stored in directory: /root/.cache/pip/wheels/df/99/da/c34f202dc8fd1dffd35e0ecf1a7d7f8374ca05fbcbaf974b83
  Building wheel for netifaces (setup.py): started
  Building wheel for netifaces (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /opt/conda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-g7bu0azh
       cwd: /tmp/pip-install-9px94vhh/netifaces/
  Complete output (16 lines):
  running bdist_wheel
  running build
  running build_ext
  checking for getifaddrs...not found.
  checking for getnameinfo...not found.
  checking for socket IOCTLs...not found.
  checking for optional header files...none found.
  checking whether struct sockaddr has a length field...no.
  checking which sockaddr_xxx structs are defined...none!
  checking for routing socket support...no.
  checking for sysctl(CTL_NET...) support...no.
  checking for netlink support...no.
  building 'netifaces' extension
  gcc -pthread -B /opt/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DNETIFACES_VERSION=0.10.9 -I/opt/conda/include/python3.7m -c netifaces.c -o build/temp.linux-x86_64-3.7/netifaces.o
  unable to execute 'gcc': No such file or directory
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for netifaces
  Running setup.py clean for netifaces
  Building wheel for pyrsistent (setup.py): started
  Building wheel for pyrsistent (setup.py): finished with status 'done'
  Created wheel for pyrsistent: filename=pyrsistent-0.17.3-cp37-cp37m-linux_x86_64.whl size=55874 sha256=c10aefa1da62e56905c6d2bbc6269614343a65b02bde636d1b1e13aed5ed6808
  Stored in directory: /root/.cache/pip/wheels/a5/52/bf/71258a1d7b3c8cbe1ee53f9314c6f65f20385481eaee573cc5
  Building wheel for configobj (setup.py): started
  Building wheel for configobj (setup.py): finished with status 'done'
  Created wheel for configobj: filename=configobj-5.0.6-py3-none-any.whl size=34546 sha256=41dbe3b816c4524e0c001d7f84aa55a0d6f74615e30f1eb9e036ce2c4cd7ef44
  Stored in directory: /root/.cache/pip/wheels/0d/c4/19/13d74440f2a571841db6b6e0a273694327498884dafb9cf978
Successfully built wrapt nvidia-ml-py3 pyrsistent configobj
Failed to build netifaces
Installing collected packages: wurlitzer, requests-toolbelt, attrs, pyrsistent, typing-extensions, zipp, importlib-metadata, jsonschema, wrapt, nvidia-ml-py3, configobj, everett, dulwich, netifaces, websocket-client, comet-ml
    Running setup.py install for netifaces: started
    Running setup.py install for netifaces: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /opt/conda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-o13vmqqd/install-record.txt --single-version-externally-managed --compile --install-headers /opt/conda/include/python3.7m/netifaces
         cwd: /tmp/pip-install-9px94vhh/netifaces/
    Complete output (16 lines):
    running install
    running build
    running build_ext
    checking for getifaddrs...not found.
    checking for getnameinfo...not found.
    checking for socket IOCTLs...not found.
    checking for optional header files...none found.
    checking whether struct sockaddr has a length field...no.
    checking which sockaddr_xxx structs are defined...none!
    checking for routing socket support...no.
    checking for sysctl(CTL_NET...) support...no.
    checking for netlink support...no.
    building 'netifaces' extension
    gcc -pthread -B /opt/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DNETIFACES_VERSION=0.10.9 -I/opt/conda/include/python3.7m -c netifaces.c -o build/temp.linux-x86_64-3.7/netifaces.o
    unable to execute 'gcc': No such file or directory
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/conda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9px94vhh/netifaces/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-o13vmqqd/install-record.txt --single-version-externally-managed --compile --install-headers /opt/conda/include/python3.7m/netifaces Check the logs for full command output.
The command '/bin/sh -c pip install comet-ml' returned a non-zero code: 1
```

**Comment by dsblank** [comment]
Sorry to hear about weather! 

Looks like netifaces needs the gcc tools to build. I see that you are in a conda environment. You can also use our conda package (and conda channel) [1] or install the tools to install gcc [2].

[1] - https://www.comet.ml/docs/python-sdk/releases/ - "conda install -c anaconda -c conda-forge -c comet_ml comet_ml"
[2] - https://stackoverflow.com/questions/29732990/installing-a-gcc-compiler-onto-a-docker-container

**Comment by david-wb** [comment]
@dsblank Ok, but seems odd that comet-ml would depend on gcc tools. `wandb` Installs without a hitch in the pytorch and tensorflow base docker images and I think comet-ml should too as it will be commonly used in pytorch or tensorflow containers.

**Comment by dsblank** [comment]
We'll take a look and see what this dependency is. Thanks for reporting.

**Comment by dsblank** [comment]
Should be fixed with comet_ml version 3.5.0: https://www.comet.ml/docs/python-sdk/releases/#release-350
